job/src/main/java/org/jenkinsci/plugins/workflow/job/console/WorkflowMetadataConsoleFilter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.job.console;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import hudson.console.LineTransformationOutputStream;
+
+/**
+ * It transforms workflow metadata log messages through {@link WorkflowRunConsoleNote}.
+ */
+public class WorkflowMetadataConsoleFilter extends LineTransformationOutputStream {
+
+    private final OutputStream out;
+
+    public WorkflowMetadataConsoleFilter(OutputStream out) {
+        this.out = out;
+    }
+
+    @Override
+    protected void eol(byte[] b, int len) throws IOException {
+        new WorkflowRunConsoleNote().encodeTo(out);
```

Comment:
Well, the note object is directly serialized in the log so I guess the must be different objects, but I'm not sure, perhaps someone with a deeper knowledge of the `ConsoleNote` API could tell us?

BTW this is the pattern I've seen everywhere when using console notes (for example [here](https://github.com/jenkinsci/openJDK-native-plugin/blob/4ef6b93b56b3d7ca896c787aad04671fd6d1c87d/src/main/java/org/jenkinsci/plugins/openjdk_native/OpenJDKConsoleAnnotator.java#L23), [here](https://github.com/jenkinsci/grails-plugin/blob/8e1aafa0482272d981f681e7eec8d82221ecfac0/src/main/java/com/g2one/hudson/grails/GrailsBuilder.java#L264) or [here](https://github.com/jenkinsci/docker-build-step-plugin/blob/3c2295f3a54c6fecda4e705bd0b87c108f015993/src/main/java/org/jenkinsci/plugins/dockerbuildstep/log/DockerConsoleAnnotator.java#L31)).